lib/pure/os.nim

```diff
@@ -3310,3 +3310,61 @@ func isValidFilename*(filename: string, maxLen = 259.Positive): bool {.since: (1
     find(f.name, invalidFilenameChars) != -1): return false
   for invalid in invalidFilenames:
     if cmpIgnoreCase(f.name, invalid) == 0: return false
+
+when defined(windows):
```

Comment:
looks like it attempts to do same thing as nativesockets.getHostname but with sublte differences:

```nim
proc getHostname*(): string {.tags: [ReadIOEffect].} =
  ## Returns the local hostname (not the FQDN)
  # https://tools.ietf.org/html/rfc1035#section-2.3.1
  # https://tools.ietf.org/html/rfc2181#section-11
  const size = 64
  result = newString(size)
  when useWinVersion:
    let success = winlean.gethostname(result, size)
  else:
    # Posix
    let success = posix.gethostname(result, size)
  if success != 0.cint:
    raiseOSError(osLastError())
  let x = len(cstring(result))
  result.setLen(x)
```

- [ ] why 64 vs 256?
- [ ] why not use winlean.gethostname instead of getComputerNameExA?

if the implementations end up with same semantics, you could simply have it defined in os.nim and re-exported in nativesockets via:
```nim
export getHostname
```

Reply:
> why not use winlean.gethostname instead of getComputerNameExA

The latter is better because it doesn't require a WSAStartup call.